clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -581,8 +581,16 @@ Expected<StringRef> clang(ArrayRef<StringRef> InputFiles, const ArgList &Args) {
   if (SaveTemps)
     CmdArgs.push_back("-save-temps");
 
-  if (SaveTemps && linkerSupportsLTO(Args))
-    CmdArgs.push_back("-Wl,--save-temps");
+  if (linkerSupportsLTO(Args)) {
+    if (SaveTemps)
+      CmdArgs.push_back("-Wl,--save-temps");
+    if (Args.hasArg(OPT_lto_debug_pass_manager))
+      CmdArgs.push_back("-Wl,--lto-debug-pass-manager");
```

Comment:
The `--save-temps` thing is kind of special, I'm thining for stuff like this it's better suited for `-Xoffload-linker --lto-debug-pass-manager` once I merge https://github.com/llvm/llvm-project/pull/101032.